src/sonic-vrrpcfgd/vrrpcfgd/vrrpcfgd.py

```diff
@@ -0,0 +1,727 @@
+#!/usr/bin/env python
+
+import copy
+import subprocess
+import time
+import syslog
+import os
+from swsscommon.swsscommon import ConfigDBConnector
+import socket
+import threading
+import queue
+import signal
+import re
+import logging
+import netaddr
+import ipaddress
+import io
+import struct
+
+class CachedDataWithOp:
+    OP_NONE = 0
+    OP_ADD = 1
+    OP_DELETE = 2
+    OP_UPDATE = 3
+
+    STAT_SUCC = 0
+    STAT_FAIL = 1
+
+    def __init__(self, data = None, op = OP_NONE):
+        self.data = data
+        self.op = op
+        self.status = self.STAT_FAIL
+
+    def __repr__(self):
+        op_str = ''
+        if self.op == self.OP_NONE:
+            op_str = 'NONE'
+        elif self.op == self.OP_ADD:
+            op_str = 'ADD'
+        elif self.op == self.OP_DELETE:
+            op_str = 'DELETE'
+        elif self.op == self.OP_UPDATE:
+            op_str = 'UPDATE'
+        return '(%s, %s)' % (self.data, op_str)
+
+vrrpd_client = None
+
+def g_run_command(table, command, daemons, ignore_fail = False):
+    syslog.syslog(syslog.LOG_DEBUG, "execute command {} for table {}.".format(command, table))
+    if not command.startswith('vtysh '):
+        syslog.syslog(syslog.LOG_ERR, 'Command no start with "vtysh". Command: "{}"'.format(command))
+    else:
+        if not vrrpd_client.run_vtysh_command(table, command, daemons) and not ignore_fail:
+            syslog.syslog(syslog.LOG_ERR, 'command execution failure. Command: "{}"'.format(command))
+            return False
+    return True
+
+def extract_cmd_daemons(cmd_str):
+    # daemon list could be given within brackets at head of input lines
+    dm_mark = re.match(r'\[(?P<daemons>.+)\]', cmd_str)
+    if dm_mark is not None and 'daemons' in dm_mark.groupdict():
+        cmd_str = cmd_str[dm_mark.end():]
+        daemons = dm_mark.groupdict()['daemons'].split(',')
+    else:
+        daemons = None
+    return (daemons, cmd_str)
+
+class VrrpdClientMgr(threading.Thread):
+    VTYSH_MARK = 'vtysh '
+    PROXY_SERVER_ADDR = '/etc/frr/vrrpd_client_sock'
+    ALL_DAEMONS = ['vrrpd']
+    TABLE_DAEMON = {
+            'VRRP': ['vrrpd'],
```

Comment:
I believe, we discussed to have VRRP config handler for FRR as part of frrcfgd itself, what are we doing here?

Reply:
Yes, we have also implemented this functionality in frrcfgd. This provides an alternative for usage when frrcfgd is not enabled. Some users may need to use it this way.

Reply:
vrrcfgd is totally a duplicate code, I would like to better understand in which case vrrpcfgd will be enabled but not frrcfgd. Keeping both the code will be hard to manage, I would recommend using frrcfgd only for VRRP feature.

Reply:
> vrrcfgd is totally a duplicate code, I would like to better understand in which case vrrpcfgd will be enabled but not frrcfgd. Keeping both the code will be hard to manage, I would recommend using frrcfgd only for VRRP feature.

OK, vrrpcfgd has been removed, thanks.